src/modules/cpu/kernel/resample.hpp

```diff
@@ -0,0 +1,149 @@
+                    Rpp32f inPos = inBlockRaw - inBlockRounded;
+                    const Rpp32f *inBlockPtr = srcPtrTemp + inBlockRounded;
+                    for (int outPos = outBlock; outPos < blockEnd; outPos++, inPos += fscale)
+                    {
+                        Rpp32s loc0, loc1;
+                        window.input_range(inPos, &loc0, &loc1);
+                        if (loc0 + inBlockRounded < 0)
+                            loc0 = -inBlockRounded;
+                        if (loc1 + inBlockRounded > srcLength)
+                            loc1 = srcLength - inBlockRounded;
+                        Rpp32s locInWindow = loc0;
+                        Rpp32f locBegin = locInWindow - inPos;
+                        __m128 pLocInWindow = _mm_add_ps(_mm_set1_ps(locBegin), xmm_pDstLocInit);
+
+                        Rpp32f accum = 0.0f;
+                        __m128 pAccum = xmm_p0;
+                        for (; locInWindow + 3 < loc1; locInWindow += 4)
+                        {
+                            __m128 w4 = window(pLocInWindow);
+                            pAccum = _mm_add_ps(pAccum, _mm_mul_ps(_mm_loadu_ps(inBlockPtr + locInWindow), w4));
+                            pLocInWindow = _mm_add_ps(pLocInWindow, xmm_p4);
+                        }
+                        // sum all 4 values in the pAccum vector and store in accum
+                        pAccum = _mm_add_ps(pAccum, _mm_shuffle_ps(pAccum, pAccum, _MM_SHUFFLE(1, 0, 3, 2)));
+                        pAccum = _mm_add_ps(pAccum, _mm_shuffle_ps(pAccum, pAccum, _MM_SHUFFLE(0, 1, 0, 1)));
+                        accum = _mm_cvtss_f32(pAccum);
+
+                        Rpp32f x = locInWindow - inPos;
+                        for (; locInWindow < loc1; locInWindow++, x++) {
+                            Rpp32f w = window(x);
+                            accum += inBlockPtr[locInWindow] * w;
+                        }
+                        dstPtrTemp[outPos] = accum;
+                    }
+                }
+            }
+            else
+            {
+                std::vector<Rpp32f> tempBuf;
+                tempBuf.resize(numChannels);
+                for (int outBlock = outBegin; outBlock < outEnd; outBlock += block)
+                {
+                    Rpp32s blockEnd = std::min(outBlock + block, outEnd);
+                    Rpp64f inBlockRaw = outBlock * scale;
+                    Rpp32s inBlockRounded = static_cast<int>(inBlockRaw);
+                    Rpp32f inPos = inBlockRaw - inBlockRounded;
+                    const Rpp32f *inBlockPtr = srcPtrTemp + inBlockRounded * numChannels;
+                    for (int outPos = outBlock; outPos < blockEnd; outPos++, inPos += fscale)
+                    {
+                        Rpp32s loc0, loc1;
+                        window.input_range(inPos, &loc0, &loc1);
+                        if (loc0 + inBlockRounded < 0)
```

Comment:
Is this (loc0 + inBlockRounded) or loc0 - inBlockRounded ?
How can it become -ve?